backend/apps/vacancies/models.py

```diff
@@ -0,0 +1,131 @@
+        about vacancy
+    priority : str
+        priority in vacancies
+    salary_max : int
+        maximum salary
+    min_max : int
+        minimum salary
+    author : str
+        author by vacancy
+    contact_person : str
+        contact person y vacancy
+    is_active : boolean
+        if vacancy is active
+    is_salary_show : boolean
+        if we need in vacancy show salary
+    """
+
+    title = models.CharField(_("Title"), max_length=64, db_index=True)
+    keywords = TaggableManager()
+    type_of_employment = models.CharField(
+        _("Type of employment"), choices=EmploymentChoices.choices, db_index=True, max_length=32
+    )
+    location = models.CharField(_("Location"), max_length=128, db_index=True)
+    english_level = models.CharField(
+        _("English level"), choices=EnglishLevelChoices.choices, db_index=True, max_length=32
+    )
+    min_experience = models.FloatField(_("Minimum experience in years"), db_index=True, blank=True)
+    end_date = models.DateField(_("Recruitment end date"))
+    start_date = models.DateField(_("Recruitment start date"))
+    description = models.TextField(_("Description"), max_length=2048)
+    priority = models.TextField(_("Priority"), max_length=512)
+    salary_max = models.IntegerField(_("Maximum salary"), db_index=True)
+    salary_min = models.IntegerField(_("Minimum salary"), db_index=True)
+    salary_currency = models.ForeignKey(Currency, on_delete=models.DO_NOTHING, db_index=True)
+    author = models.ForeignKey(User, on_delete=models.DO_NOTHING, related_name="vacancy_author")
+    contact_person = models.ForeignKey(
+        User, on_delete=models.DO_NOTHING, related_name="vacancy_contact_person"
+    )
+    is_active = models.BooleanField(_("Is active"), default=False)
+    is_salary_show = models.BooleanField(_("Is salary show"), default=False)
+
+    class Meta:
+        ordering = ("-id",)
+
+    def __str__(self):
+        return self.title
+
+    @property
+    def salary(self):
+        if self.salary_min == self.salary_max:
+            return f"{self.salary_max} {self.salary_currency.currency_code}"
+        return f"{self.salary_min}-{self.salary_max} {self.salary_currency.currency_code}"
+
+    @property
+    def show_min_experience(self):
+        if self.min_experience:
+            return f"{self.min_experience} years"
+        return "without min experience"
+
+    @property
+    def search_period(self):
+        if self.start_date == self.end_date:
+            return f"...-{self.end_date.strftime('%Y.%m.%d')}"
+        return f"{self.start_date.strftime('%Y.%m.%d')}-{self.end_date.strftime('%Y.%m.%d')}"
+
+    def clean(self):
+        if self.end_date < self.start_date:
+            raise ValidationError(
+                _("The end date of the search cannot be earlier than the start date!")
+            )
+        if self.salary_max < self.salary_min:
+            raise ValidationError(_("The minimum salary cannot be higher than the maximum!"))
```

Comment:
I don't think it's a good idea to validate the fields inside the `clean()` method because it will work only when you create a model instance.
The django documentation says: "a model’s clean() method is not invoked when you call your model’s save() method"
(https://docs.djangoproject.com/en/4.1/ref/models/instances/#django.db.models.Model.clean)
We should validate these fields inside the form or serializer.
(https://stackoverflow.com/a/36413343/15390079)

Reply:
But we can simply call the clean method already in the serializer

Reply:
Such way will be implicit for the whole team because everyone must remember to call `model.clean()` method all the time they need to `save` the instance inside all of the serializer's methods or anywhere else...
Serializers have their own `validate()` method which exactly designed for this.
Keep it simple! Imagine if you leave the project, a new developer comes on a board and he/she has no idea about that specific rule (always call `clean()` method before saving a vacancy instance).